### PR TITLE
Add Controller to check classification of snippets

### PR DESCRIPTION
**Pull request by coolo**

### CODE REVIEW

File: lib/Cavil/Model/Snippets.pm [hunk 1]
```diff
@@ -35,4 +35,14 @@ sub find_or_create {
   return $db->select('snippets', 'id', {hash => $hash})->hash->{id};
 }
 
+sub random {
+  my ($self, $limit) = @_;
+
+  return $self->pg->db->query(
+    'select id, text, classified,
+         license, confidence from snippets where approved=FALSE
+        order by random() limit ?', $limit
```
Comment (coolo) [inline lib/Cavil/Model/Snippets.pm]: this seems not to work performant for larger tables.

**Comment by coolo** [comment]
I didn't put any user checks on it as I would like to broaden this a little. But I'm afraid we need a new role for it.

File: lib/Cavil/Model/Snippets.pm (hunk 1 above)
Comment (kraih) [reply lib/Cavil/Model/Snippets.pm]: This is faster, but you need to know the number of rows in the table. 🤔
```sql
select id, text, classified, license, confidence from snippets where approved=false offset floor(random()*427040) limit 100;
```

Comment (kraih) [reply lib/Cavil/Model/Snippets.pm]: Luckily getting the highest id is a very fast query (unlike count).
```sql
select max(id) from snippets;
```

**Comment by kraih** [comment]
A new role would make sense.

File: lib/Cavil/Model/Snippets.pm (hunk 1 above)
Comment (coolo) [reply lib/Cavil/Model/Snippets.pm]: then we better have no gaps in the ids.

Comment (kraih) [reply lib/Cavil/Model/Snippets.pm]: Getting the estimate is also very fast.
```sql
select reltuples as approximate_row_count from pg_class where relname = 'snippets';
```

Comment (coolo) [reply lib/Cavil/Model/Snippets.pm]: ordering by (indexed) hash proved to be fast and random enough :)

**Comment by coolo** [comment]
Added classifier role (added it to all admins for now), added test case, added help text.

**Comment by coolo** [comment]
Ok, ordering by hash isn't good enough after all. It's random to the user, but we really should aim for true randomness:

- if a user is not sure about one specific snippet, reloading the page will avoid that decision
- two users working in parallel will review different things (at least quite likely)

**Pull request merged**